include/bluetooth/audio.h

```diff
@@ -1172,10 +1172,82 @@ struct bt_audio_broadcast_sink_cb {
 
 /** @brief Stream operation. */
 struct bt_audio_stream_ops {
+	/** @brief Stream configured callback
+	 *
+	 *  Configured callback is called whenever an Audio Stream has been
+	 *  configured.
+	 *
+	 *  @param stream Stream object that has been configured.
+	 */
+	void (*configured)(struct bt_audio_stream *stream);
+
+	/** @brief Stream QoS set callback
+	 *
+	 *  QoS set callback is called whenever an Audio Stream Quality of
+	 *  Service has been set or updated.
+	 *
+	 *  @param stream Stream object that had its QoS updated.
+	 */
+	void (*qos_set)(struct bt_audio_stream *stream);
+
+	/** @brief Stream enabled callback
+	 *
+	 *  Enabled callback is called whenever an Audio Stream has been
+	 *  enabled.
+	 *
+	 *  @param stream Stream object that has been enabled.
+	 */
+	void (*enabled)(struct bt_audio_stream *stream);
+
+	/** @brief Stream started callback
+	 *
+	 *  Started callback is called whenever an Audio Stream has been started
+	 *  and will be usable for streaming.
+	 *
+	 *  @param stream Stream object that has been started.
+	 */
+	void (*started)(struct bt_audio_stream *stream);
+
+	/** @brief Stream metadata updated callback
+	 *
+	 *  Metadata Updated callback is called whenever an Audio Stream's
+	 *  metadata has been updated.
+	 *
+	 *  @param stream Stream object that had its metadata updated.
+	 */
+	void (*metadata_updated)(struct bt_audio_stream *stream);
+
+	/** @brief Stream disabled callback
+	 *
+	 *  Disabled callback is called whenever an Audio Stream has been
+	 *  disabled.
+	 *
+	 *  @param stream Stream object that has been disabled.
+	 */
+	void (*disabled)(struct bt_audio_stream *stream);
+
+	/** @brief Stream stopped callback
+	 *
+	 *  Stopped callback is called whenever an Audio Stream has been
+	 *  stopped.
+	 *
+	 *  @param stream Stream object that has been stopped.
+	 */
+	void (*stopped)(struct bt_audio_stream *stream);
+
+	/** @brief Stream released callback
+	 *
+	 *  Released callback is called whenever a Audio Stream has been
+	 *  released and can be deallocated.
+	 *
+	 *  @param stream Stream object that has been released.
+	 */
+	void (*released)(struct bt_audio_stream *stream);
+
 	/** @brief Stream connected callback
 	 *
 	 *  If this callback is provided it will be called whenever the
-	 *  connection completes.
+	 *  isochronous stream is usable.
```

Comment:
I would rather say "becomes ready to use", or "becomes usable"
As for "whenever", that seems to indicate that is happens many times.  Could that be replace with "when"?

Most importantly: Clarify the difference between this callback and the "started" callback - both are currently saying that they are called when the stream is "usable". 
Or am I just missing the distinction between "stream" and "isochronous stream"?
Should this callback's @brief be with "Isochronous stream connected callback"?  (Ditto below.)

Reply:
Good feedback. 

This callback was added before the `started` callback, and essentially the actual callback that an application should use would be the `started` callback, and not this, as any data would be dismissed until the remote endpoint is in the `streaming` state (which triggers the `started` callback). The ISO stream shall be connected before the remote endpoint can enter this state. 

Can you think of a scenario where the upper layer may want to know whether the ISO is connected on its own? i.e. should we remove the stream connected and disconnected callbacks?

Reply:
Is the upper layer ever doing anything with the ISO?
The abstraction for the upper layer is the (audio) stream, and as long as the ISO is handled invisibly to the upper layers, it should be OK to remove them.

Reply:
It is not. 

The callbacks were added as we didn't have any way earlier to determine whether or not we could actually start sending data, i.e. the client would do `bt_audio_stream_start()` and once that was accepted and completed, it would work, but there was no callback for it. It worked fine for the shell (primarily using debug messages), but not for proper applications. 

However, these may still prove useful for the broadcast sink, as there is no `start` and `stop` callback for that. We could reuse the same callbacks for broadcast. 

What do you think?

Reply:
So - keep the iso stream callbacks and use the for the broadcast sink?
How about rather wrapping them to the same abstraction level (audio stream) as for unicast?

Reply:
We could remove the connected and disconnected ones, and then keep the started and stopped for broadcast sink (possible also for the broadcast source), and then keep the remaining for unicast only.

Reply:
Sounds like a good solution to me.